app.py

```diff
@@ -1,10 +1,122 @@
-from flask import Flask
+from flask import Flask, jsonify, request
+import traceback
+import random
+import datetime
+from test_mongo_data import Mdb
+from config import ITEM, CONTAINER, ARRANGEMENT, SNAPSHOT
+import json
+from bson import ObjectId
+
 app = Flask(__name__)
+mdb = Mdb()
+
+class JSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, ObjectId):
+            return str(o)
+        return json.JSONEncoder.default(self, o)
+
+def uniqueid():
+    seed = random.getrandbits(32)
+    while True:
+        yield seed
+        seed += 1
 
 
-@app.route('/')
-def hello():
+@app.route("/")
+def home_page():
     return "Arrange That!"
 
+
+@app.route('/arrangement', methods=['POST'])
+@app.route('/api/v1/arrangement', methods=['POST'])
+def save_arrangement():
+    arrangement = request.json
+    json_data = validate_arrangement(arrangement)
+    if json_data == True:
+        arrangement_exists = mdb.check_arrangement_exists(arrangement)
+        if arrangement_exists:
+            mdb.replace_arrangement(arrangement)
+        else:
+            mdb.add_arrangement(arrangement)
+        return JSONEncoder().encode(arrangement)
```

Comment:
why not use jsonify instead of JSONEncoder?

Reply:
if JSONEncoder is better, can we just use that instead of jsonify?

Reply:
let me think about that.

Reply:
I am not sure if it matters at this point: here's my reference 
https://realpython.com/python-json/

To translate a custom object into JSON, all you need to do is provide an encoding function to the class  method’s default parameter. The json module will call this function on any objects that aren’t natively serializable.

Reply:
ok that's fine then